support/ref-docs/src/documenter.ts

```diff
@@ -0,0 +1,624 @@
+    // Prepend the new children
+    (firstPara as Parent).children.unshift(...children);
+  }
+
+  return nodes as Node[];
+}
+
+/**
+ * Attempts to link to a target API item by its name.
+ */
+function tryLinkName(
+  ctx: DocumentationContext,
+  name: string,
+  child: Node
+): Node {
+  // Attempt to find the target item
+  const [target] = ctx.pkgEntry.findMembersByName(name);
+  // TODO: handle multiple targets?
+  // TODO: better extraction by using tokens instead of
+
+  // If the target isn't found, simply return the child w/o a link
+  if (target === undefined) {
+    return child;
+  }
+
+  return link(`#${target.displayName}`, undefined, child);
+}
+
+/**
+ * Formats certain types to be more human-friendly.
+ */
+function formatExcerptReferences(
+  ctx: DocumentationContext,
+  excerpt: Excerpt
+): Node[] {
+  // Bail early for object types/interface shorthand
+  if (excerpt.text.match(/^{(?:\s|\S)*}(?:\[\])*$/) !== null) {
+    // TODO: better substitution that explains properties
+    return [inlineCode('object')];
+  }
+
+  return [
+    html('<code>'),
+    ...excerpt.tokens
+      // Get the relevant tokens
+      .filter(
+        (_, i) =>
+          i >= excerpt.tokenRange.startIndex && i < excerpt.tokenRange.endIndex
+      )
+      // Map `Content`-kind tokens to just their text & `Reference`-kind tokens by
+      // their own format function
+      .map(token =>
+        token.kind === ExcerptTokenKind.Content
+          ? text(token.text)
+          : tryLinkName(ctx, token.text, text(token.text))
+      ),
+    html('</code>')
+  ];
+}
+
+/**
+ * Converts a DocSection to `mdast` nodes.
+ */
+function convertDocSection(
+  ctx: DocumentationContext,
+  section: DocSection
+): Node[] {
+  // Elements within sections are paragraphs
+  return (section.nodes as readonly DocParagraph[]).map(para =>
+    paragraph(
+      para.nodes
+        // Filter out soft breaks
+        .filter(node => !(node instanceof DocSoftBreak))
+        // Convert DocNodes to mdast nodes
+        .map(node => {
+          if (node instanceof DocPlainText) {
+            // Plain text maps to `text` nodes
+            return text(node.text);
+          } else if (node instanceof DocLinkTag) {
+            // Links map to links. The URI depends on the destination.
+            if (node.codeDestination !== undefined) {
+              // TODO: better code link resolution
+              const name = node.codeDestination.memberReferences[0]
+                .memberIdentifier!.identifier;
+              return tryLinkName(ctx, name, inlineCode(name));
+            } else if (node.urlDestination !== undefined) {
+              return link(node.urlDestination, undefined, [
+                text(
+                  node.linkText === undefined
+                    ? node.urlDestination
+                    : node.linkText
+                )
+              ]);
+            } else {
+              throw new Error('misformatted DocLinkTag');
```

Comment:
just so i understand, if you throw here, it will bubble up and terminate the documentation generation process, right?

Reply:
I don't *quite* know of what situation would cause this, but it means that it's not known how to format this link. So yes, it would terminate the doc generation process because it bubbles up.